build/deploydb.sh

```
@@ -20,17 +27,39 @@ else
-CREATE DATABASE zstack_rest;
-grant all privileges on zstack.* to root@'%' identified by "${password}";
-grant all privileges on zstack_rest.* to root@'%' identified by "${password}";
-grant all privileges on zstack.* to root@'127.0.0.1' identified by "${password}";
-grant all privileges on zstack_rest.* to root@'127.0.0.1' identified by "${password}";
+# Detect MySQL version
+# Extract major version number from various MySQL/MariaDB/GreatDB output formats
+db_version=$(${MYSQL} --version 2>/dev/null | grep -oP '\d+\.\d+\.\d+' | head -1 | cut -d'.' -f1 || echo "5")
+
+# GreatDB and MySQL 8.0+ require CREATE USER before GRANT
+if command -v greatdb &> /dev/null || [ "$db_version" -ge 8 ] 2>/dev/null; then
+  ${MYSQL} ${loginCmd} << EOF
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)